packages/connectivity/src/scp-cf/destination/destination-from-registration.ts

```diff
@@ -0,0 +1,140 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import { Cache, IsolationStrategy } from '../cache';
+import { decodeJwt } from '../jwt';
+import { userId } from '../user';
+import { getXsuaaServiceCredentials } from '../environment-accessor';
+import { parseSubdomain } from '../subdomain-replacer';
+import { Destination, DestinationAuthToken } from './destination-service-types';
+import { DestinationFetchOptions } from './destination-accessor-types';
+import { DestinationCache } from './destination-cache';
+import {
+  addProxyConfigurationInternet,
+  ProxyStrategy,
+  proxyStrategy
+} from './proxy-util';
+
+const logger = createLogger({
+  package: 'connectivity',
+  messageContext: 'register-destination'
+});
+/**
+ * @internal
+ */
+export const registerDestinationCache = DestinationCache(
+  new Cache<Destination>(undefined)
+);
+
+type RegisterDestinationOptions = Pick<
+  DestinationFetchOptions,
+  'jwt' | 'isolationStrategy'
+>;
+
+/**
+ * Registers a destination in a cache for later usage.
+ *
+ * Throws an error if a destination with the same name as the given test destination already exists.
+ * @param destination - A destination to add to the `destinations` cache
+ * @param options - Options how to cache the destination
+ */
+export function registerDestination(
+  destination: DestinationWithName,
+  options?: RegisterDestinationOptions
+): void {
+  if (!destination.name || !destination.url) {
+    throw Error(
+      'Registering destinations requires a destination name and url.'
+    );
+  }
+
+  registerDestinationCache.cacheRetrievedDestination(
+    decodedJwtOrZid(options),
+    destination,
+    isolationStrategy(options)
+  );
+}
+
+export type DestinationWithName = Omit<Destination, 'name'> & { name: string };
```

Comment:
```suggestion
export type DestinationWithName = Destination & { name: string };
```
This should make it required.